### PR TITLE
Add invariance tests

### LINKED ISSUE #38: testing invariances
In the paper, we claim several types of invariance for the methods implemented here:

- Invariance to column fixed effects (for all methods): Re-mapping Yit <- Yit + bt for arbitrary bt doesn't change anything (either weights or CI estimates).
- Invariance to row fixed effects (for DID and SDID): Re-mapping Yit <- Yit + ai for arbitrary ai doesn't change anything (either weights or CI estimates).
- Invariance to scaling (all methods): Re-mapping Yit <- c * Yit for some c > 0 doesn't change weights, and makes the treatment effect estimate be multiplied by c. (For numerical robustness, it'd be nice to check that this goes through with both very big and very small c, e.g., c = 10^{+/- 6}.)

It would be valuable to have tests that check that these invariances are enforced.

In some cases, specific choices of tuning parameters may interfere with invariance. In this case, however, we should make sure that all methods with defaults (i.e., without tuning parameters explicitly specified) respect the invariances.

### PR DESCRIPTION
**Pull request by erikcs**
This should resolve #38

### CODE REVIEW

**Comment by erikcs** [comment]
Tangential issue: when writing this the interface was a bit confusing, having the `estimate` being a double with many attributes attached

I don't see the big benefit over just dressing the return object up as a list like in grf?

One could have `coef(estimate)` return the `tau.hat`, and `coef(estimate, type = "weights")` return weights suitable for a user (and `summary` could print the coefficient output along with std.error instead of having synthdid_se exported?

**Comment by davidahirshberg** [comment]
There isn't really a good reason to have it implemented that way, but we should probably keep it as is until we do some kind of interface-breaking release. But we should implement and use implementation-independent accessors as you propose.  Re: standard errors, they should be programatically accessible as opposed to only available via summary. Given that we have 3 standard error estimators to implement, we should probably use a function like synthdid_se (or to simplify naming, a new generic 'se') rather than shoving them into predict(est, 'se') like some packages, but I could be convinced otherwise.

**Comment by erikcs** [comment]
Sounds good, it does not seem R has any consistent pattern for standard errors, they have `vcov`, we can figure out something idiomatic later on

**Comment by erikcs** [comment]
pinging @swager @davidahirshberg

File: tests/testthat/test_synthdid.R [hunk 1]
```diff
@@ -66,3 +40,63 @@ test_that("default synthdid behavior has not changed", {
...
+    weights = attr(estimate, "weights")
+    estimate.scaled = estimator(c.small * setup$Y, setup$N0, setup$T0)
+    estimate.se.scaled = synthdid_se(estimate.scaled)
+    weights.scaled = attr(estimate.scaled, "weights")
+    expect_equal(c(c.small * estimate), c(estimate.scaled))
+    expect_equal(c.small * estimate.se, estimate.se.scaled)
+    expect_equal(weights[c("lambda", "omega")], weights.scaled[c("lambda", "omega")])
+  }
+  # 3.2: c is very small
+  c.large = 10^-6
```
Comment (swager) [inline tests/testthat/test_synthdid.R:90]: large vs small?

**Review by swager — APPROVED** [review]
Looks good, thanks!

**Pull request merged**